homeassistant/components/auth/__init__.py

```diff
@@ -262,28 +271,32 @@ def __init__(self, retrieve_credentials):
         grant_type = data.get('grant_type')
 
         if grant_type == 'authorization_code':
-            return await self._async_handle_auth_code(hass, client_id, data)
+            return await self._async_handle_auth_code(
+                hass, request, client_id, data)
 
         elif grant_type == 'refresh_token':
             return await self._async_handle_refresh_token(
-                hass, client_id, data)
+                hass, request, client_id, data)
```

Comment:
Thinking here, instead of `return await`, what if we do something like this:

```python
resp = await self._async_handle_refresh_token(…)
if resp.status != 200:
    await process_wrong_login(request)
```

In this case we know something is off if the status is not 200?

Reply:
oops! 😢